### PR TITLE
tests, vm_test: Remove tests that verifies basic k8s functionality

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Following discussion at https://github.com/kubevirt/kubevirt/pull/13015/files#r1789759957

Remove tests following tests because they verify basic k8s functionality, not related to Kubevirt:
- "[test_id:232]should create same manifest twice via command line" Any k8s type or CRD type instance can be created multiple times
- "[test_id:233][posneg:negative]should fail when deleting nonexistent VM" Any non existing type instance deletion should fail with not-exist error from k8s API server.
- "[test_id:243][posneg:negative]should create VM only once" Any type instance that create twice should fail with an 'already-exist' error by k8s API server.
- Tests under Context "as ordinary OCP user trough test service account" These tests verifies some 'oc' functionality, which is not an artifact by this project.

Before this PR:

After this PR:

Fixes #
https://github.com/kubevirt/kubevirt/issues/12709

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
Good start! I'm not convinced we need the remaining kubectl test cases either. These tests are a mixture of kubectl and the golang client. Can these cases be covered by only using the golang client? (if necessary at all)

**Comment by 0xFelix** [comment]
Fixes https://github.com/kubevirt/kubevirt/issues/12709

**Review by dankenigsberg — APPROVED** [review]

File: tests/vm_test.go
Comment (xpivarc) [inline tests/vm_test.go:998]: +1, fyi @fossedihelm

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by ormergi** [comment]
> Good start! I'm not convinced we need the remaining kubectl test cases either. These tests are a mixture of kubectl and the golang client. Can these cases be covered by only using the golang client? (if necessary at all)

I wasn't sure about removing the remaining ones, maybe you guys can help me figure out if there are similar tests that cover the same scenarios but using the go-client:

- `[release-blocker][test_id:299]should create VM via command line using all supported API versions`
   Verifying the VM CRD version v1alpha4 and v1 works
   
Do we need to care about v1alpha4?

- `[test_id:264]should create and delete a VM`
This test create, delete a VM and verifies its pods is deleted as well, using kubectl.

Is there a test that cover this scenario elsewhere?

- Tests under context `should not change anything if dry-run option is passed`
Tests virt-client dry-run feature, do you know if there are other tests that cover dry-run?

> Fixes #12709
> 
> Can you add this to the PR description?

Done

**Comment by ormergi** [comment]
@0xFelix @xpivarc @dankenigsberg I posted PR to change remaining tests to use go-client.
https://github.com/kubevirt/kubevirt/pull/13029

**Comment by ormergi** [comment]
/hold

Placing hold until we have ack from @0xFelix, in regard to https://github.com/kubevirt/kubevirt/pull/13026#pullrequestreview-2354861683

**Comment by 0xFelix** [comment]
/hold cancel

This PR looks fine to me. Thanks.

**Pull request merged**
